scalameta/src/main/scala/scala/meta/internal/parsers/Parsers.scala

```diff
@@ -2911,7 +2966,7 @@ private[meta] class Parser(val input: Input)(implicit val dialect: Dialect) { pa
    */
   def blockStatSeq(): List[Stat] = {
     val stats = new ListBuffer[Stat]
-    while (!token.is[StatSeqEnd] && !token.is[CaseDefEnd]) {
+    while (!token.is[StatSeqEnd] && !token.is[CaseDefEnd] && !ahead(token.is[`case`])) {
```

Comment:
But isn't `!ahead(token.is[`case`])` going to be false, which will prompt an early exit from the loop?